scripts/textmesh/text-mesh.js

```diff
@@ -0,0 +1,397 @@
+/* jshint esversion: 6 */
+const BEZIER_STEP_SIZE = 3;
+
+const EPSILON = 1e-6;
+
+const p = new pc.Vec2();
+const p1 = new pc.Vec2();
+const p2 = new pc.Vec2();
+
+const vTemp1 = new pc.Vec2();
+const vTemp2 = new pc.Vec2();
+const vTemp3 = new pc.Vec2();
+const vTemp4 = new pc.Vec2();
+const vTemp5 = new pc.Vec2();
+const vTemp6 = new pc.Vec2();
```

Comment:
Should these variables be scoped so they don't pollute the global space?

Reply:
I can make them static on the `Polygon` class but that'd just make the code a fair bit more verbose. I know it's not ideal, but let's keep it like this for now. At some point in the future, it'd be good to move the `Polygon` API into the engine and then this won't be a problem.